### PR TITLE
fix nav, add dynamic routing, start on support pages

### PR DESCRIPTION
**Pull request by by-bai**
**stuff done:**
- added assets folder by designers to the project, as raised in meeting on 8 dec '21
- fixed navigation for bottom navbar - tab remains active when subpages are visited. there were some bugs previously
   - the following code is required in subpages of a feature:
   - ` <BottomNav name={feature name} />` 
   - e.g. ` <BottomNav name="support" />` for subpages under support feature
- started working on support pages. components and pages for this feature are under `components/support` & `pages/support` folders respectively
  - linked up main support page to subcategory pages (try clicking on a category icon to be directed to subcategory page which displays its subcategories)
  - added dynamic routing for subcategory page `/support/{subcategory}` e.g. `/support/career`, `/support/financial` etc.
  - completed basic render of necessary icons and images for each category

**to do:**

continue working on UI for support feature. lots of styling and UI behaviour are incomplete in comparison with current figma designs; purpose of this PR is to sync up bottom navigation and asset imports from designers so that shiqi can start working on the profile feature :)

### CODE REVIEW

**Comment by by-bai** [comment]
update: fixed broken CI build :')

File: components/support/Category.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  { type: 'family', image: family, text: 'Family Services' },
+  { type: 'children', image: children, text: 'Children Education' },
+  { type: 'legal', image: legal, text: 'Legal' }
+]
+
+export default function Category(props: Props) {
+  const type = props.type
+  const category_ref = category_list.filter(
+    (category) => category.type === type
+  )[0]
```
Comment (myyk) [inline components/support/Category.tsx:31]: I think we should probably check to see that the's an element first.

If there's no element we should log an error and skip it.

**Review by myyk — CHANGES_REQUESTED** [review]

File: components/support/Category.tsx [hunk 2]
```diff
@@ -14,28 +14,31 @@ interface Props {
...
+      : category_dict[`${type}`]
 
   return (
     <div>
-      <Link href={`/support/${category_ref.type}`}>
+      <Link href={`/support/${type}`}>
         <div>
-          <Image src={category_ref.image} />
-          <p className="text-body-main">{category_ref.text}</p>
+          <Image src={category_ref[0]} />
```
Comment (myyk) [inline components/support/Category.tsx:40]: Won't this still have some sort or problem if `category_ref` does not exist? I think so, but also I'm not that familiar with react. This is where a test would help.

File: pages/support/[subcategory].tsx [hunk 3]
```diff
@@ -69,8 +71,8 @@ export default function Support() {
     <div className={styles.container}>
       {}
       <TopNav pageName={'Support Subcategory'} displayBackButton={true} />
-      <Image src={category_ref.image} />
-      <h4>{category_ref.text}</h4> {/*  TODO: change weight to semibold */}
+      <Image src={category_ref[0]} />
```
Comment (myyk) [inline pages/support/[subcategory].tsx:74]: Same problem with error handling here.

File: components/support/Category.tsx [hunk 4]
```diff
@@ -14,28 +14,31 @@ interface Props {
...
-    (category) => category.type === type
-  )[0]
+  const category_ref =
+    category_dict[`${type}`] == null
+      ? console.log(
+          'error in Category.tsx: category_ref is undefined as category type does not exist.'
+        )
+      : category_dict[`${type}`]
 
   return (
```
Comment (myyk) [inline components/support/Category.tsx]: I think the simplest way to handle this error would be something like this (sorry if my typescript syntax is incorrect):

```
if (category_dict[`${type}`] == null) {
    console.log('error in Category.tsx: category_ref is undefined as category type does not exist.')
} else {
    return (
       <div>
       ...
    }
}
```

**Review by myyk — CHANGES_REQUESTED** [review]

File: components/support/Category.tsx (hunk 4 above)
Comment (by-bai) [reply components/support/Category.tsx]: ahh i understand now, will push the changes soon. i'm not familiar with writing tests but will look into that too. thanks!

**Comment by by-bai** [comment]
error handling update: log error and `return null` (nothing is rendered) if there's an error. 

screenshot:
image

what should the user see if this happens in production?

**Review by myyk — APPROVED** [review]
Yeah, I'm also not sure what's the best practice here. But it's good to see it not completely crashed when this happens.

**Comment by myyk** [comment]
> what should the user see if this happens in production?

I'm not sure either, designers definitely won't be able to plan for every type of error that's possible. There's a way to catch errors and display an error when they are unhandled with Error Boundaries. But I think that's for when things are uncaught mostly. If they are anticipated then we get to do whatever we thing is smarter.

Like if we were displaying N things, but 1 of them wasn't in one of those dicts, it would be best if we discovered this during our release process. But if we couldn't bc it's configured on our backend, then just showing N-1 working ones, would be much better. Then somehow getting that error back up to us.

That's how I'd guess would be a good way to do it.

**Pull request merged**